### ISSUE TITLE
Question regarding KL Divergence implementation (Forward vs Reverse) and Figure 1 calculation

### ISSUE THREAD

**Issue by cth2888** [zhuchichi56/ASFT#3]
Hi, thank you for the code and the paper. I have two questions regarding the KL divergence:

1. KL Calculation in Figure 1：Could you clarify the exact formula used for the "KL divergence" curve in Figure 1? Is it calculated on the training data or generated rollouts, and is it Forward or Reverse KL?

2. Forward vs. Reverse KL： in CodeThe paper (Section 5.3) states that Reverse KL ($D_{KL}(Q \parallel P)$) is used to anchor the model.However, in train.py, the loss is implemented as:
Pythonkl_div = F.kl_div(
    F.log_softmax(shift_logits, dim=-1),  # input: log(Q)
    F.softmax(ref_logits, dim=-1),        # target: P
    reduction="none",
).sum(dim=-1)

According to [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.nn.functional.kl_div.html), kl_div(input, target) computes $\sum target \cdot (\log target - input)$.Since target here is the reference model ($P$), this code computes Forward KL ($D_{KL}(P \parallel Q)$), which corresponds to mode-covering behavior.Is this discrepancy between the code (Forward KL) and the paper (Reverse KL) intended?

Thanks!
